### ISSUE TITLE
Jobs page performance

### ISSUE THREAD

**Issue by gisjedi** [ngageoint/scale#27]
When a number of jobs have built up the response time of the Jobs list page seriously degrades.  I'm guessing this is a result of using a sort on the update time of the job combined with paging.  This is going to cause the entire jobs table to be processed by the database so that the appropriate page of data can be returned.

A simple solution would be to add a days/time back filter to ensure nothing older than a week or so would be processed.  This could be exposed as a calendar control and passed to the API.  This would reduce the total dataset that would need to be processed when viewing the job list.

**Comment by dfaller** [comment]
There is already a datetime range available in the REST API. This just needs a UI update.

**Comment by dickmancj-mil** [comment]
This seems like an indexing and query optimization issue. We can try the datetime range filter on site tomorrow but this feels like a workaround.

**Comment by dickmancj-mil** [comment]
Did not mean to close this.

**Comment by cshamis** [comment]
@gisjedi: That was it.  We were missing an index on last_modified_date in the recipe and jobs tables.  Now django and Postgres can do the paging correctly and the performance is very good.

**Comment by dickmancj-mil** [comment]
The Django model needs to be updated to include the index. Changed the labels. IMO this should be done this sprint.

**Comment by dfaller** [comment]
We still need the datetime range. Eric has it in place, he's just doing some tweaking to its location. Chris and I think a week makes a decent default range.

**Comment by dfaller** [comment]
Oh, we should also put the same widget on the recipe page as well.

**Comment by dfaller** [comment]
I can get the indexes added to the code this sprint. We'll have to remember to remove the ones Clint made when we bring the code over.

**Comment by dickmancj-mil** [comment]
This item was for performance, which is critical. The date range filter should be opened as a separate issue and discussed with GIAT team whether it's necessary or useful.

**Comment by ericsvendsen-mil** [comment]
The date filter has already been removed from the develop branch due to discussions I had with GIAT personnel yesterday afternoon. If we do reinstate it, there should not be a default range selected that limits the total records returned. We can discuss this further, but I agree that it should be a separate issue.

**Comment by dfaller** [comment]
The filter is necessary for performance. It limits the number of records which get counted. As the number of jobs/recipes increase over time, the pages are going to get increasingly slower. Yes, I imagine the index helps right now but it doesn't solve the longer term issue.

**Comment by cshamis** [comment]
I agree, performance is critical. And although the new indexes VASTLY improve the responsiveness, the version in dev with the 30 day date filter returns instantaneously, where prod (also with new indexes) still takes up to 3-4 seconds.  More importantly though, when additionally filtering by status or job-type, we're back to our old 10-15 second wait... And that's only going to keep getting slower as more and more jobs are added.  The key here to remember is, Scale doesn't have a "rolling" history, it just keeps adding on forever.

Moreover though, I'm just not sure there's a use case for the jobs page (by default) return every job since the beginning of time.  If (as a user) i want that behavior, I'll can always just set the calendar to 1970 and go forward...  Otherwise, I probably just to see the last 72 hours worth.  

The benefits are: It reduces the table scans load on the DB, it reduces the response times, and it means that sorts with combined filters can be applied to a MUCH smaller result result set which again makes for damn near instantaneous page loads.   Which --in government software-- never happens... So it makes it seem even FASTER!

-C.

**Comment by dfaller** [comment]
Has the datetime widget been added back in? If not, can you get it added? Either way, if the datetime widget is getting in this sprint, can you guys add a commit that closes this issue?
